spec/factories/ems_folder.rb

```diff
@@ -19,27 +19,27 @@
   # VMware specific folders
   #
 
-  factory :vmware_folder, :parent => :ems_folder do
+  factory :vmware_folder, :parent => :ems_folder, :class => "ManageIQ::Providers::Vmware::InfraManager::Folder" do
```

Comment:
if the other vmware_folder objects inherit from this one, then you can set the class once.